File: src/main/java/me/kaotich00/fwauctionhouse/storage/sql/ConnectionFactory.java
package me.kaotich00.fwauctionhouse.storage.sql;

import me.kaotich00.fwauctionhouse.FwAuctionHouse;

import java.sql.Connection;
import java.sql.SQLException;

public interface ConnectionFactory {

    void init(FwAuctionHouse plugin);

    void shutdown() throws Exception;

    Connection getConnection() throws SQLException;

}

File: src/main/java/me/kaotich00/fwauctionhouse/storage/sql/ConnectionFactory.kt
package me.kaotich00.fwauctionhouse.storage.sql

import me.kaotich00.fwauctionhouse.FwAuctionHouse
import java.sql.Connection

interface ConnectionFactory {

    fun init(plugin: FwAuctionHouse)

    fun shutdown()

    val connection: Connection
}